src/components/Header/components/SearchButton.tsx

```diff
@@ -6,24 +6,34 @@ import Button from "../../Button/Button";
 import Icon from "../../Icons/Icon";
 import SearchForm from "./SearchForm";
 
-const SearchButton = chakra(() => {
+interface SearchButtonProps {
+  isMobile?: boolean;
+}
+
+const SearchButton = chakra(({ isMobile = false }: SearchButtonProps) => {
   const [isOpen, setIsOpen] = useState<boolean>(false);
-  const styles = useStyleConfig("HeaderSearchButton", { isOpen });
+  const styles = useStyleConfig("HeaderSearchButton", { isMobile, isOpen });
+  const buttonText = isMobile ? null : isOpen ? "Close" : "Search";
 
   return (
     <FocusLock isDisabled={!isOpen}>
       <Button
         aria-haspopup="true"
+        aria-label={isOpen ? "Close Search" : "Open Search"}
         aria-expanded={isOpen ? true : null}
         buttonType="link"
         id="searchButton"
         onClick={() => setIsOpen(!isOpen)}
         __css={styles}
       >
-        {isOpen ? "Close" : "Search"}
-        <Icon align="right" name={isOpen ? "close" : "search"} size="small" />
+        {buttonText}
+        <Icon
+          align={isMobile ? "none" : "right"}
+          name={isOpen ? "close" : "search"}
+          size={isMobile ? "large" : "medium"}
```

Comment:
The other icons in the mobile view are size="medium". Do you think we should increase the size of all of them?